### DISCUSSION TITLE
A button to go to the previous step the user saw?

### DISCUSSION THREAD

**Discussion by tettoffensive** [shipshapecode/shepherd#2485] [Q&A] [answered]
I have a tour which has multiple paths. I see that `back` goes the the previous step in the tour by looking at the previous index in the array. But I have 2 different steps which have buttons `return this.show('learn-more');` then in my learn more step I have a "back" button that I want to go to the step which the user came from.

**Comment by crudobaker** [comment, marked as answer]
Hey @tettoffensive 

I think you should control the tour state by your own. I did something similar by subscribing to the **show** event of the tour. That listener receives the current step you are on. So you can keep a reference of that step in your own TourState and by doing that, you can also save the previous state.

Hope it helps.

**Reply by tettoffensive** [reply]
Thanks. I think I'll do that, but thought it would be something built-in.

**Reply by RobbieTheWagner** [reply]
There is nothing built in that tracks the last step that was seen, so keeping track in your app is best.

**Comment by crudobaker** [comment]
Hey @tettoffensive 

I kept thinking in your use case, because maybe I will face the same in the near future.

I'd try to avoid tours with different paths at tour designing time. You can have one instance of the `learn-step` and reuse it in any tour declaration.

**Reply by tettoffensive** [reply]
Thanks. I think I'll keep track of the state on my own. I was not a part of the design time on this one but the "learn more" tour step is kind of like temporarily jumping out of the tour to another dialog and then the user may choose to go back to the tour. I had to ship something quickly so I decided to use a tour step rather than implement my own dialog.
